### ISSUE TITLE
Style Property of Components

### ISSUE THREAD

**Issue by georgeduckett** [stoiveyp/Alexa.NET.APL#37]
Shouldn't the `Style` property of Components be a string, since according to their documentation it is supposed to refer to a named style, rather than it being the style object directly. Currently I'm resorting to `CustomerComponent` with a custom `style` property which is a string value.

Is there something I've missed?

**Comment by stoiveyp** [comment]
All APLValue properties can be assigned directly from a string, so `component.Style = "myStyle1"` should work just fine (this is because the property needs to be able to be generated from an expression).

That said, Style should _also_ be updated as it does appear the rules on that have changed in the last few months 👍

**Comment by georgeduckett** [comment]
Thanks for the quick response. It doesn't look like it's working here (3.8.0 NuGet). It says it can't convert from `string` to `APLValue<Style>`, which is correct; there's code to go from `T` to `APLValue<T>` but not from `string`.

**Comment by stoiveyp** [comment]
sorry - quick because I wasn't on my laptop. Instead of a custom expression you can use:

```csharp
component.Style = APLValue.To<Style>("style");
```

or longhand

```csharp
component.Style = new APLValue<Style>{Expression="style"};
```

both easier to maintain short term rather than customising components.

**Comment by georgeduckett** [comment]
Ahh, got it, thanks! Thanks for taking the time to explain.

**Comment by stoiveyp** [comment]
FYI - This has just been fixed in 3.9.0
